docs/datamodel/functions.rst

```diff
@@ -32,6 +32,67 @@ This function accepts a :eql:type:`str` as an argument and produces a
   test> select exclamation({'Hello', 'World'});
   {'Hello!', 'World!'}
 
+.. _ref_datamodel_functions_modifying:
+
+Modifying Functions
+^^^^^^^^^^^^^^^^^^^
+
+.. versionadded:: 6.0
+
+User-defined functions can contain DML (i.e.,
+:ref:`insert <ref_eql_insert>`, :ref:`update <ref_eql_update>`,
+:ref:`delete <ref_eql_delete>`) to make changes to existing data. These
+functions have a :ref:`modifying <_ref_reference_volatility>` volatility.
+
+.. code-block:: sdl
+
+  function add_user(name: str) -> User
+    using (
+      insert User {
+        name := name,
+        joined_at := std::datetime_current(),
+      }
+    );
+
+.. code-block:: edgeql-repl
+
+    db> select add_user('Jan') {name, joined_at};
+    {default::User {name: 'Jan', joined_at: <datetime>'2024-12-11T11:49:47Z'}}
+
+Unlike other functions, the arguments of modifying functions **must** have a
+:ref:`cardinality <_ref_reference_cardinality>` of ``One``.
```

Comment:
Practical question: if some property that you're trying to insert or update is `multi`, does that mean you'll have to pass it as an array and unpack it? Maybe we should show this as an example?